src/nv_ingest/util/nim/helpers.py

```diff
@@ -258,3 +389,44 @@ def is_ready(http_endpoint, ready_endpoint) -> bool:
         # Don't let anything squeeze by
         logger.warning(f"Exception: {ex}")
         return False
+
+
+def get_version(http_endpoint, metadata_endpoint="/v1/metadata", version_field="version") -> str:
+    if http_endpoint is None or http_endpoint == "":
+        return ""
+
+    url = generate_url(http_endpoint)
+    url = remove_url_endpoints(url)
+
+    if not metadata_endpoint.startswith("/") and not url.endswith("/"):
+        metadata_endpoint = "/" + metadata_endpoint
+
+    url = url + metadata_endpoint
+
+    # Call the metadata endpoint of the NIM
+    try:
+        # Use a short timeout to prevent long hanging calls. 5 seconds seems resonable
+        resp = requests.get(url, timeout=5)
```

Comment:
I think we may want to think about some number of retries here, and even a standard for when we consider a NIM request to be completely failed. It definitely seems possible that a NIM could be down or unresponsive for a short time and then come back. Do we really want to fail the job if this happens?

Reply:
Thanks for catching this. A standard for when we consider a NIM to have completely failed sounds like a good idea, but honestly I'm not sure exactly how that looks like. For now, I just added a simple retry on the function in https://github.com/NVIDIA/nv-ingest/pull/154/commits/caa4de888936239b2b3ce8e2c0066ddf41ae2fa8.